packages/block-library/src/navigation/style.scss

```diff
@@ -19,25 +24,192 @@
 	margin: 0;
 	padding-left: 0;
 
-	// Horizontal layout
-	display: flex;
-	flex-wrap: wrap;
+	// Only hide the menu by default if responsiveness is active.
+	.is-responsive {
+		display: none;
+	}
 
-	// Vertical layout
-	.is-vertical & {
-		display: block;
+	@include break-small() {
+		// Horizontal layout
+		display: flex;
+		flex-wrap: wrap;
+
+		// Vertical layout
+		.is-vertical & {
+			display: block;
+		}
 	}
 }
 
 // Justification.
-.items-justified-center > ul {
+// These target the named container class to work even with the additional mobile menu containers.
+.items-justified-center .wp-block-navigation__container {
 	justify-content: center;
 }
 
-.items-justified-right > ul {
+.items-justified-right .wp-block-navigation__container {
 	justify-content: flex-end;
 }
 
-.items-justified-space-between > ul {
+.items-justified-space-between .wp-block-navigation__container {
 	justify-content: space-between;
 }
+
+
+/**
+ * Mobile menu.
+ */
+
+.wp-block-navigation__responsive-container {
+	display: none;
+	position: fixed;
+	top: 0;
+	left: 0;
+	right: 0;
+	bottom: 0;
+	background-color: #fff;
+	/*
+		arbitrary, high z-index
+		@todo: find correct value in order to make sure
+		that this is always on top.
+	*/
+	z-index: 1;
+	align-items: flex-start;
+	justify-content: flex-start;
+
+	// Provide an opinionated default style for menu items inside.
+	// Inherit as much as we can regarding colors, fonts, sizes,
+	// but otherwise provide a baseline.
+	// In a future version, we can explore more customizability.
+	&.is-menu-open {
+		display: flex;
+		flex-direction: column;
+
+		// Give it a z-index just higher than the adminbar.
+		z-index: 100000;
```

Comment:
Happy to do so. I intentionally didn't because this is the frontend style, and I'm trying to figure out how much of the block editor principia to bring over there. For example I try to avoid variables like `$gray-900`, because _these are theme styles_ and should be considered in light of all themes.

Reply:
Oh right, frontend! 🤔 Seems okay to leave it as is then